library/std/src/macros.rs

```diff
@@ -379,3 +379,51 @@ macro_rules! dbg {
         ($($crate::dbg!($val)),+,)
     };
 }
+
+/// Creates a [`HashMap`] containing the arguments.
+///
+/// This macro creates or either an empty [`HashMap`] using
+/// [`HashMap::new`] if there are no arguments, or generates
+/// a [`HashMap::insert`] call for each element.
+///
+/// This macro allows trailing commas.
+///
+/// This macro is used by concatenating comma separated sequences
+/// of `<key:expr> => <value:expr>`.
+///
+/// An example usage of this macro could be the following
+///
+/// ```rust
+/// #![feature(hash_map_macro)]
+///
+/// let map = hash_map!{
+///     "key" => "value",
+///     "key1" => "value1"
+/// };
+///
+/// for (key, value) in map {
+///     println!("{key:?} => {value:?}");
+/// }
+/// ```
+///
+/// Note that since this macro only generates [`HashMap::insert`]
+/// calls, the move semantics for the values are the same as
+/// the insert method.
+///
+/// [`HashMap`]: crate::collections::HashMap
+/// [`HashMap::new`]: crate::collections::HashMap::new
+/// [`HashMap::insert`]: crate::collections::HashMap::insert
+#[macro_export]
+#[allow_internal_unstable(hint_must_use)]
```

Comment:
you don'takw use of this feature (maybe it would make sense to?)

Reply:
(Noratrieb's point was that you actually have to use `must_use` within the macro. This line only makes it so that using `must_use` within the macro is *allowed*.)